### PR TITLE
refactor(source): extract and merge common fields of connector source and table source

### PR DESCRIPTION
**Pull request by xx01cyx**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Previously in `catalog.proto`,  both `StreamSourceInfo` and `TableSourceInfo` contain fields `row_id_index`, `columns`, `pk_column_ids` and `properties`, which should actually belong to `Source` instead of `XxxSourceInfo`:

```proto
// catalog.proto

message StreamSourceInfo {
  map<string, string> properties = 1;
  plan_common.RowFormatType row_format = 2;
  string row_schema_location = 3;
  ColumnIndex row_id_index = 4;
  repeated plan_common.ColumnCatalog columns = 5;
  repeated int32 pk_column_ids = 6;
}

message TableSourceInfo {
  ColumnIndex row_id_index = 1;
  repeated plan_common.ColumnCatalog columns = 2;
  repeated int32 pk_column_ids = 3;
  map<string, string> properties = 4;
}

message Source {
  ...
  oneof info {
    StreamSourceInfo stream_source = 5;
    TableSourceInfo table_source = 6;
  }
  ...
}
```

This PR extracts the fields to `Source` proto (also adds some comments):

```proto
// catalog.proto

message StreamSourceInfo {
  plan_common.RowFormatType row_format = 1;
  string row_schema_location = 2;
}

message TableSourceInfo {}

message Source {
  ...
  // The column index of row ID. If the primary key is specified by the user, this will be `None`.
  ColumnIndex row_id_index = 5;
  // Columns of the source.
  repeated plan_common.ColumnCatalog columns = 6;
  // Column id of the primary key specified by the user. If the user does not specify a primary key, the vector will be empty.
  repeated int32 pk_column_ids = 7;
  // Properties specified by the user in WITH clause.
  map<string, string> properties = 8;
  // Information of either stream source (connector) or table source.
  oneof info {
    StreamSourceInfo stream_source = 9;
    TableSourceInfo table_source = 10;
  }
  ...
}
```

and refactors rust code accordingly.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

**Review by tabVersion — APPROVED** [review]
LGTM

**Review by yezizp2012 — APPROVED** [review]
LGTM

File: src/common/src/catalog/column.rs [hunk 1]
```diff
@@ -54,6 +55,26 @@ impl From<ColumnId> for i32 {
...
+    }
+}
+
+impl From<ColumnId> for ProstColumnIndex {
+    fn from(column_id: ColumnId) -> Self {
+        Self {
+            index: column_id.0 as _,
+        }
+    }
+}
```
Comment (st1page) [inline src/common/src/catalog/column.rs:76]: looks like a dangerous behavior...

File: src/frontend/src/catalog/source_catalog.rs [hunk 2]
```diff
@@ -38,6 +40,8 @@ pub struct SourceCatalog {
     pub append_only: bool,
     pub owner: u32,
     pub info: SourceCatalogInfo,
+    pub row_id_index: Option<ColumnId>,
```
Comment (st1page) [inline src/frontend/src/catalog/source_catalog.rs:43]:
```suggestion
    pub row_id_index: Option<usize>,
```

**Review by st1page — COMMENTED** [review]
rest LGTM

File: src/common/src/catalog/column.rs (hunk 1 above)
Comment (BugenZhao) [reply src/common/src/catalog/column.rs:76]: +1. We should not confuse ID with index, in favor of table alternating in the future. 🤔

**Pull request merged**